feat: load_all_pages improvments

AhmedHodiani opened the pull request:
- Added `next_token_key` Parameter: Allows customization of the key used to retrieve the next_token from the response, supporting different API structures e.g. the latest finance API version.

- Improved Next Token Handling: Checks both the top-level response and payload for the next_token using the specified key in `next_token_key`. (for some reason the top-level token is always null, where the token inside the payload is not. in the new finance api version) 

- Updated Documentation: Clarified the use of the new next_token_key parameter in the docstring.

----
`next_token_key` default is `next_token` just like the old behavior.
both `next_token_key` and `next_token_param` should set to `nextToken` when working with Finance V_2024_06_19

## Summary by Sourcery

Enhance the load_all_pages utility to support different API structures by allowing customization of the key used to retrieve the next token from the response. This change also improves next token handling by checking both the top-level response and payload for the next token.

Enhancements:
- Add the `next_token_key` parameter to allow customization of the key used to retrieve the next_token from the response.
- Improve next token handling by checking both the top-level response and payload for the next_token using the specified key in `next_token_key`.
- Update documentation to clarify the use of the new `next_token_key` parameter.

saleweaver commented:
Hi! Thanks for that. Ideally, we would add this to the ApiResponse class, where it can be used even when load all pages is not used. By fixing ApiResponse to the newly introduced possibility of passing the token by amazon, it'll also be available in load_all_pages.

Have a look here: https://github.com/saleweaver/python-amazon-sp-api/blob/master/sp_api/base/ApiResponse.py

AhmedHodiani commented:
hello @saleweaver, thanks for reviewing. I reverted the changes on load_all_pages and added the new token possibility to ApiResponse as you suggested.
